internal/jobcontainers/jobcontainer.go

```diff
@@ -31,6 +31,15 @@ import (
 	"golang.org/x/sys/windows"
 )
 
+var fileBindingSupport bool
+
+func init() {
+	bindDLL := `C:\windows\system32\bindfltapi.dll`
```

Comment:
Would there be a need to explicitly disable bind support, even if the API is available? Ie, if someone didnt want their paths mucked with or their process started in a command prompt?

Reply:
Agree that even if we can't use Hamza's approach, we should at least do this in a `sync.Once` or something so we only check it if we need it. Otherwise we'll be doing filesystem access every time this package is imported.

Reply:
I'd imagine most folks would want the new behavior, but semi related I like the idea of an escape hatch to the old way if there was a bug so I should probably add an annotation regardless. The DLL is actually loaded that way via the winapi bindings but it's much easier to setup this branching old/new logic if we know we have the functionality available instead of "try and call functionality -> fail/succeed -> swap approach". 

>> Otherwise we'll be doing filesystem access every time this package is imported.

I'd never actually tested this, but is that true? I did a quick test of 
```go
package foo

func init() {
	fmt.Println("hi")
}
```

and did a blank import of foo on two other packages that all get imported in main and only end up seeing one "hi" to stdout.

Reply:
You are correct that a given Go process will only import the same package once, regardless of how many packages depend on it.

The issue is not that it gets imported more than once in a given process, it's that implicitly doing filesystem access just because someone referenced your package (which in the future could be just because they want to use some constant you defined, for instance), is bad.

Reply:
Gotcha, I misunderstood what you were implying with that. I was kind of lax with the init approach as we're the only ones importing this, but even for our case this stat is useless if we're launching the shim for a Windows Server Container for example

Reply:
Swapped to sync.Once on the first container create, let me know if this looks alright